### ISSUE TITLE
3.10 hangs while rendering a component that references a property with its own name

### ISSUE THREAD

**Issue by backspace** [emberjs/ember.js#18215]
Hey, I found that while updating an application from Ember 3.9 to 3.11, one of the routes wouldn’t render at all and I had to kill the browser. I tracked it down to a component rendering a computed property that had the same name as itself and [fixed it](https://github.com/backspace/prison-rideshare-ui/pull/114/commits/8297fde396f38b70c3285d2308801f40bb036b5f), but I thought I’d let you all know in case this is considered a breaking change, or could somehow be detected and warned about. It seems related to #18094, but having nothing render could be a confusing experience.

I made [this repository](https://github.com/backspace/component-property-clash) to reproduce in a more limited context, starting . In [this commit](https://github.com/backspace/component-property-clash/commit/8e53d20a7a931b060622296844e6cbff730e97a9) I added a component called `clashing-name` with this implementation:

```
export default Component.extend({
  clashingName: computed('text', function() {
    return this.text;
  }),
});
```

The template just references that computed property:

```
<div class="clashing-name">{{clashingName}}</div>
```

On Ember 3.9, all is well, [the tests pass](https://travis-ci.org/backspace/component-property-clash/builds/561767910#L633).

When I [updated to Ember 3.10](https://github.com/backspace/component-property-clash/commit/47cccb6c41722828e6d7e45f3d1c752fd2462ba3), [the tests failed](https://travis-ci.org/backspace/component-property-clash/jobs/561768988#L502-L506) with a timeout.

It’s a simple fix; changing the template to have `{{this.clashingName}}` works, or changing the name of the property to something else. My concern is more that someone with less familiarity might have a hard time finding what the problem is, as there’s no indication what’s going on, the browser just becomes unresponsive.

Let me know if I can provide any more information that might be helpful. Thanks for all the Octane work, I’ve been trying it in another application and it’s great!

**Comment by rwjblue** [comment]
Yikes! Sorry about the issues here. I suspect the underlying issue is that we removed the dash requirement for ambiguous mustaches (the `{{clashingName}}` in your example) in 3.10. In 3.9 `{{clashingName}}` is only going to be a helper invocation or property lookup, but in 3.10 it is also allowed to be component invocation.

I'm not sure what exactly we can do here....

**Comment by rwjblue** [comment]
Actually, I think we can fix it over in ember-resolver. This section needs to be updated to exclude `component` in addition to `helper`:

https://github.com/ember-cli/ember-resolver/blob/c44257496a5280ceed407832822fe4b90a43989f/addon/resolvers/classic/index.js#L176

**Comment by backspace** [comment]
It wasn’t a big deal to narrow down and fix, using the classic technique of deleting templates and restoring them piece by piece, I’m just more thinking about people who wouldn’t necessarily think of that approach. But I appreciate you looking into it. I could try a PR if it’s helpful, I’m guessing a test [here](https://github.com/ember-cli/ember-resolver/blob/master/tests/unit/resolvers/classic/basic-test.js) would be warranted?

**Comment by rwjblue** [comment]
Yep, exactly what I was thinking.

**Comment by rwjblue** [comment]
Released in ember-resolver@5.2.0: https://github.com/ember-cli/ember-resolver/releases/tag/v5.2.0

Closing this one now that I think we've addressed the underlying issue.
